src/cdk/portal/portal.ts

```diff
@@ -153,6 +153,22 @@ export class TemplatePortal<C = any> extends Portal<C> {
   }
 }
 
+/**
+ * An `InlinePortal` is a portal whose content will be taken from its current position
+ * in the DOM and moved into a portal outlet, when it is attached. On detach, the content
+ * will be restored to its original position.
+ */
+export class InlinePortal extends Portal<HTMLElement> {
```

Comment:
What do you think about `DomPortal` or `ElementPortal`?

cc @josephperrott and @mmalerba for opinions

Reply:
Both `DomPortal` and `ElementPortal` sound good to me.